framework/doc/content/source/auxkernels/FunctorCoordinatesFunctionAux.md

```diff
@@ -0,0 +1,31 @@
+# FunctorCoordinatesFunctionAux
+
+!syntax description /AuxKernels/FunctorCoordinatesFunctionAux
+
+The `FunctorCoordinatesFunctionAux` lets users pass variables, functions, postprocessors and other [Functors](Functors/index.md) as time/coordinates arguments to a function, in order to set an auxiliary variable. It is generally useful when the function shape is only available as a [Function](Functions/index.md), even
+though it would often be more appropriate to use a material property.
+
+!alert warning
+If any of the arguments are nonlinear variables or depend on nonlinear variables, you should not
+use the `FunctorCoordinatesFunctionAux` when also using [automatic differentiation](automatic_differentiation/index.md) and Newton's method.
+Auxiliary variables cannot store derivatives with regards to the nonlinear variables, so the Jacobian for Newton's method
+would be incomplete.
+
+!alert warning
+If using auxiliary variables as arguments to the `FunctorCoordinatesFunctionAux`, you must check that the ordering of execution
+is as expected, in that the auxiliary kernels that set those auxiliary variables are executed BEFORE the `FunctorCoordinatesFunctionAux`.
+This can be done with the [!param](/Debug/show_execution_order) parameter in the `[Debug]` block.
```

Comment:
It's not enough that one aux kernel supplying an aux var needed for this new aux kernel is executed before within a given `execute_on`. It must be executed at some previous `execute_on`. This is because the aux variable functor evaluation reads from the `_sys.currentSolution` vector which, despite its name, is not updated until after aux kernels have been evaluated for a given `execute_on`. This is (somewhat) documented in https://github.com/idaholab/moose/issues/19396

Reply:
I ll add a note. it s a shame to have caveats in the functor APIs which we wanted to solve all this

Reply:
We could probably fix 99% of use cases by drawing from the parallel solution if the variable kind is auxiliary. This will be wrong if someone tries to, for instance, fill a nodal finite element variable on a processor boundary with a nodal functor evaluation of an auxiliary finite volume variable (non-local values would be wrong in this case)

Reply:
Fixing the aux kernel limitations hasn't really been a priority since functor-driven simulations such as finite volume can almost always just use a functor material property in any place an aux variable might be used. But since most finite element simulation objects index at quadrature points, being able to fill aux variables instead of using a functor material property is a worthwhile endeavor

Reply:
> it s a shame to have caveats in the functor APIs which we wanted to solve all this

Indeed. However, calculations involving functor material properties and nonlinear variables, which was all I was initially targeting with the functor concept, are still caveat free